### PR TITLE
fix: Filter CLI commands from manifest script detection

### PR DESCRIPTION
**Pull request by jlacube**
Prevents false positives when FileManifest._get_referenced_scripts() encounters CLI commands like 'spec-kitty' or 'git' in command templates.

Changes:
- Add CLI command filtering (spec-kitty, git, python, npm, etc.)
- Only include scripts from .kittify/scripts/ directory
- Ignore paths in other .kittify/ subdirectories (memory, templates)
- Add comprehensive test coverage

Fixes issue where command frontmatter like 'ps: spec-kitty agent --json' was incorrectly treated as a missing script file.

### CODE REVIEW

**Comment by robertDouglass** [comment]
Thanks @jlacube for identifying this issue and putting together a comprehensive fix with excellent test coverage!

We've merged a minimal fix directly to main that addresses the same problem with a simpler approach:

**The root cause**: `scripts.add(script_path)` was outside the `.kittify/` path check, so ANY first word from frontmatter (like `spec-kitty`, `git`, `python`) was added to the scripts list.

**Our fix** (3 lines): Only include paths that start with `.kittify/scripts/`:
```python
if script_path.startswith('.kittify/scripts/'):
    script_path = script_path.replace('.kittify/', '', 1)
    scripts.add(script_path)
```

This approach:
- Filters out all CLI commands automatically (no hardcoded list needed)
- Filters out other `.kittify/` subdirectories (memory, templates)
- Avoids the maintenance burden of keeping a CLI command list up to date

See commit 94c3050 for the full change with TDD tests.

If this doesn't fully address your use case or you have additional edge cases we missed, please feel free to open a new PR. Your thorough testing approach (294 lines of tests!) was great to see and helped validate our fix works correctly.

File: src/specify_cli/manifest.py [hunk 1]
```diff
@@ -111,9 +123,28 @@ def _get_referenced_scripts(self) -> List[str]:
...
+                                        # Must be in scripts/ subdirectory
+                                        if not script_path.startswith('scripts/'):
+                                            continue
+                                        scripts.add(script_path)
+                                    # Otherwise skip it (not a .kittify script)
+                                    continue
+                                else:
+                                    # Remove .kittify/ prefix for storage
+                                    script_path = script_path.replace('.kittify/', '', 1)
+                                    scripts.add(script_path)
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply src/specify_cli/manifest.py]: Audit 2026-09-05: the cited CLI-filtering branch was removed in the later canonical manifest-parser rewrite, so this complexity finding has no live code to change. This finding is superseded; thread resolution is blocked by repository permission.

File: tests/test_manifest_filtering.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
+"""Tests for manifest script filtering to prevent false positives."""
+
+from pathlib import Path
+import tempfile
+import shutil
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply tests/test_manifest_filtering.py]: Audit 2026-09-05: the cited import belonged to the removed manifest-filtering test module. No corresponding live import remains. This finding is superseded; thread resolution is blocked by repository permission.

File: tests/test_manifest_filtering.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+"""
+    (commands_dir / "test.md").write_text(command_content)
+    
+    # Test
+    manifest = FileManifest(kittify_dir)
+    scripts = manifest._get_referenced_scripts()
+    
+    # Verify actual scripts are included (platform-specific)
+    import platform
+    if platform.system() == 'Windows':
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply tests/test_manifest_filtering.py]: Audit 2026-09-05: the cited platform-specific import belonged to the removed manifest-filtering test module. No live import remains. This finding is superseded; thread resolution is blocked by repository permission.

File: tests/test_manifest_filtering.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+"""
+    (commands_dir / "cmd2.md").write_text(cli_command_content)
+    
+    # Test
+    manifest = FileManifest(kittify_dir)
+    scripts = manifest._get_referenced_scripts()
+    
+    # Verify only real script is included
+    import platform
+    if platform.system() == 'Windows':
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply tests/test_manifest_filtering.py]: Audit 2026-09-05: the cited platform-filter branch was replaced by the current canonical parser and no longer exists in the reviewed form. This finding is superseded; thread resolution is blocked by repository permission.

File: tests/test_manifest_filtering.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
+"""Tests for manifest script filtering to prevent false positives."""
+
+from pathlib import Path
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply tests/test_manifest_filtering.py]: Audit 2026-09-05: the cited Path import belonged to the removed manifest-filtering test module. No live import remains. This finding is superseded; thread resolution is blocked by repository permission.

File: src/specify_cli/manifest.py [hunk 6]
```diff
@@ -111,8 +123,24 @@ def _get_referenced_scripts(self) -> List[str]:
...
+                                    continue
+                                
+                                # Remove .kittify/ prefix for storage
+                                script_path = script_path.replace('.kittify/', '', 1)
+                                
+                                # Must be in scripts/ subdirectory
+                                if not script_path.startswith('scripts/'):
+                                    continue
+                                
                                 scripts.add(script_path)
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply src/specify_cli/manifest.py:150]: Audit 2026-09-05: the current parser deliberately accepts canonical POSIX repository-relative script paths and no longer carries the cited CLI-filter logic. This finding is superseded; thread resolution is blocked by repository permission.

File: tests/test_manifest_filtering.py [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+        assert "scripts/real-script.ps1" in scripts
+        assert len(scripts) == 1
+    else:
+        # On non-Windows, the ps: line is ignored, so no scripts
+        assert len(scripts) == 0
+    
+    # Verify CLI commands are always excluded regardless of platform
+    assert "spec-kitty" not in scripts
+    # Also verify script basename is not present as CLI command
+    assert "real-script.ps1" not in [s for s in scripts if not s.startswith('scripts/')]
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply tests/test_manifest_filtering.py]: Audit 2026-09-05: the cited redundant assertion was in the removed manifest-filtering test module. This finding is superseded; thread resolution is blocked by repository permission.

File: src/specify_cli/manifest.py [hunk 8]
```diff
@@ -111,8 +123,24 @@ def _get_referenced_scripts(self) -> List[str]:
...
+                                if not script_path.startswith('.kittify/'):
+                                    # Not a .kittify-managed script
+                                    continue
+                                
+                                # Remove .kittify/ prefix for storage
+                                script_path = script_path.replace('.kittify/', '', 1)
+                                
+                                # Must be in scripts/ subdirectory
+                                if not script_path.startswith('scripts/'):
+                                    continue
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply src/specify_cli/manifest.py]: Addressed in PR #3879 (commit bbb426def): the canonical parser now rejects backslash-containing script references instead of ambiguously transforming them. Valid script paths use canonical POSIX repository-relative syntax; 62 focused manifest and dashboard tests pass.

File: tests/test_manifest_filtering.py [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+    scripts = manifest._get_referenced_scripts()
+    
+    # Verify actual scripts are included (platform-specific)
+    if platform.system() == 'Windows':
+        assert "scripts/setup.ps1" in scripts
+    else:
+        assert "scripts/helper.sh" in scripts
+    
+    # Should have exactly one script (for current platform)
+    assert len(scripts) == 1
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply tests/test_manifest_filtering.py]: Audit 2026-09-05: the cited platform-filter assertions were in the removed manifest-filtering test module. The current parser has no corresponding platform-filter branch. This finding is superseded; thread resolution is blocked by repository permission.

File: tests/test_manifest_filtering.py [hunk 10]
```diff
@@ -0,0 +1,296 @@
...
+    scripts = manifest._get_referenced_scripts()
+    
+    # Verify backslash paths are normalized and scripts are included
+    if platform.system() == 'Windows':
+        assert "scripts/helper.ps1" in scripts
+        assert len(scripts) == 1
+    else:
+        # On non-Windows, the sh: line should also work after normalization
+        # but since the file doesn't exist as .sh, may be excluded
+        # The important thing is no errors occur
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply tests/test_manifest_filtering.py]: Audit 2026-09-05: the cited no-op platform branch was removed with the former manifest-filter implementation. This finding is superseded; thread resolution is blocked by repository permission.

File: src/specify_cli/manifest.py [hunk 11]
```diff
@@ -86,6 +91,14 @@ def _get_referenced_scripts(self) -> List[str]:
...
         is_windows = platform.system() == 'Windows'
         script_key = 'ps:' if is_windows else 'sh:'
+        
+        # Known CLI commands to exclude (not script files)
+        cli_commands = {
+            'spec-kitty', 'git', 'python', 'python3', 'pip', 'pip3',
+            'node', 'npm', 'npx', 'yarn', 'pnpm', 'poetry',
+            'docker', 'kubectl', 'terraform', 'ansible',
+            'curl', 'wget', 'jq', 'sed', 'awk', 'grep'
+        }
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply src/specify_cli/manifest.py]: Audit 2026-09-05: the cited comment describes a command-filter branch that no longer exists in the canonical parser. This finding is superseded; thread resolution is blocked by repository permission.

File: tests/test_manifest_filtering.py [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+    manifest = FileManifest(kittify_dir)
+    scripts = manifest._get_referenced_scripts()
+    
+    # Verify only real script is included
+    if platform.system() == 'Windows':
+        assert "scripts/real-script.ps1" in scripts
+        assert len(scripts) == 1
+    else:
+        # On non-Windows, the ps: line is ignored, so no scripts
+        assert len(scripts) == 0
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply tests/test_manifest_filtering.py]: Audit 2026-09-05: the cited platform-filter assertions were in the removed manifest-filtering test module. The current parser has no corresponding platform-filter branch. This finding is superseded; thread resolution is blocked by repository permission.

File: src/specify_cli/manifest.py [hunk 13]
```diff
@@ -111,8 +123,24 @@ def _get_referenced_scripts(self) -> List[str]:
                             script_parts = script_line.split()
                             if script_parts:
                                 script_path = script_parts[0]
-                                if script_path.startswith('.kittify/'):
-                                    script_path = script_path.replace('.kittify/', '')
+                                
+                                # Skip CLI commands and system executables
+                                if script_path in cli_commands:
+                                    continue
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply src/specify_cli/manifest.py]: Audit 2026-09-05: the cited CLI-command filter was removed in the canonical parser rewrite. Current script recognition starts with the canonical scripts prefix, so no live command-basename filter remains. This finding is superseded; thread resolution is blocked by repository permission.

File: src/specify_cli/manifest.py [hunk 14]
```diff
@@ -111,8 +124,29 @@ def _get_referenced_scripts(self) -> List[str]:
...
+                                
+                                # Remove .kittify/ prefix for storage
+                                script_path = script_path.replace('.kittify/', '', 1)
+                                
+                                # Must be in scripts/ subdirectory
+                                if not script_path.startswith('scripts/'):
+                                    continue
+                                
                                 scripts.add(script_path)
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply src/specify_cli/manifest.py]: Addressed in PR #3879 (commit bbb426def): manifest script references now reject empty, absolute, backslash-containing, dot, and parent-traversal paths. Focused manifest and dashboard coverage passes (62 tests), together with Ruff.

File: src/specify_cli/manifest.py [hunk 15]
```diff
@@ -111,8 +124,29 @@ def _get_referenced_scripts(self) -> List[str]:
...
+                                
+                                # Normalize path separators for cross-platform compatibility
+                                script_path = script_path.replace('\\', '/')
+                                
+                                # Skip CLI commands and system executables
+                                # Use basename to handle path prefixes like ./spec-kitty or /usr/bin/git
+                                script_basename = os.path.basename(script_path)
+                                if script_basename in cli_commands:
+                                    continue
+
```
[the remark this answers is not in the document]
Comment (samuelgoff) [reply src/specify_cli/manifest.py]: Audit 2026-09-05: the cited CLI-command filter was removed in the canonical parser rewrite. Current script recognition starts with the canonical scripts prefix, so matching script names are not filtered by a CLI-command list. This finding is superseded; thread resolution is blocked by repository permission.
